File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch04/C/P04-Sort0.cpp
// Ascending sort of an array
#include <stdio.h>
#include "sort.h"

int main()
{
   const int n = 6;                          // number of values to be sorted
   double x[] = {0., 30., 60., 50., 20., 10., 40.};     // array to be sorted
   int i;

   printf("Original array:\n");
   for (i=1; i<=n; i++) printf("%6.2f",x[i]); printf("\n");

   BubbleSort(x,n);

   printf("Sorted array:\n");
   for (i=1; i<=n; i++) printf("%6.2f",x[i]); printf("\n");
}

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch04/Python/P04-Sort0.py
# Ascending sort of an array
from sort import *

# main

n = 6                                         # number of values to be sorted
x = [0., 30., 60., 50., 20., 10., 40.]                   # array to be sorted

print("Original array:")
for i in range(1,n+1): print("{0:6.2f}".format(x[i]),end="")
print()

BubbleSort(x,n)

print("Sorted array:")
for i in range(1,n+1): print("{0:6.2f}".format(x[i]),end="")
print()
